### PR TITLE
Added lua / luajit in loops

### PR DESCRIPTION
**Pull request by SerJaimeLannister**
Hello I was really inspired by your project and I was recently experimenting with lua / luarocks (https://luarocks.org/modules/SerJaimeLannister/vigenere-cipher) (I created this myself for testing the std library of lua) and wanted to add lua support to your project

Three files have been added in the lua folder ie.

main.lua : this is what I felt personally that I could create in the most intuitive sense with no help from AI or any source as a beginner in lua , though this shouldn't primarily be used as a benchmark resource.

main_optimized.lua: some optimizations were made with the help of claude (mainly the for loop for creating the blank 10,000 array was replaced by table.create which I hadn't known till then) (skill Issue)

main_luajit.lua : this is meant to work with luajit (created with help of chatgpt) because I was having some issues trying to migrate main_optimized.lua to luajit.

In my benchmark using hyperfine 
I found main.lua to be the slowest , main_optimized.lua to be a little faster , and main_luajit.lua to be the fastest because I haven't actually checked the speed with the run.sh in the languages repo , I had created a run.sh using hyperfine in lua for lua testing 

Luajit

 Time (mean ± σ):     127.8 ms ±   1.6 ms    [User: 127.5 ms, System: 0.1 ms]
  Range (min … max):   125.2 ms … 131.0 ms    22 runs
  
  
 Lua_optimized (a little faster than the third example but slower than luajit but is in lua)
  Time (mean ± σ):      1.746 s ±  0.015 s    [User: 1.737 s, System: 0.001 s]
  Range (min … max):    1.732 s …  1.770 s    5 runs
 
 
 Lua (written by beginner in lua)
   Time (mean ± σ):      2.624 s ±  0.026 s    [User: 2.588 s, System: 0.004 s]
  Range (min … max):    2.591 s …  2.653 s    5 runs

### CODE REVIEW

File: loops/lua/main.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        print("Error: Input cannot be converted to a number")
+    else
+        -- Successful number conversion
+        local r = (math.random(number))
+        local a = {} -- new array
+        for i = 1, 10000 do
+            a[i] = 0
+        end
+        for i = 1, 10000 do
+            for j = 1, 10000 do
```
Comment (thomaswue) [inline loops/lua/main.lua:17]: This is missing a 0? See [https://github.com/bddicken/languages/blob/main/loops/c/code.c#L10C25-L10C31](https://github.com/bddicken/languages/blob/main/loops/c/code.c#L10C25-L10C31).

Comment (SerJaimeLannister) [reply loops/lua/main.lua]: I have added it , I think , sorry it was confusing and I mis added the number of 0's

Comment (SerJaimeLannister) [reply loops/lua/main.lua]: I haven't really checked it , checking it right now

Comment (SerJaimeLannister) [reply loops/lua/main.lua]: Okay so this has become a lot slower except the luajit part , maybe my pr wasn't so impressive after all

 bash run.sh
Benchmark 1: lua main_optimized.lua 40
1950039
1950040
1950032
1950003
1950033
1950037
1950004
1950027
  Time (mean ± σ):     18.502 s ±  0.036 s    [User: 18.255 s, System: 0.005 s]
  Range (min … max):   18.442 s … 18.532 s    5 runs


Benchmark 2: lua main.lua 40
1950003
1950023
1950022
1950028
1950022
1950007
1950014
1950039
  Time (mean ± σ):     26.830 s ±  0.184 s    [User: 26.515 s, System: 0.011 s]
  Range (min … max):   26.685 s … 27.152 s    5 runs

Benchmark 3: luajit main_luajit.lua 40
1950032
1950038
1950025
1950002
1950008
1950001
1950004
1950012
  Time (mean ± σ):      1.312 s ±  0.019 s    [User: 1.296 s, System: 0.001 s]
  Range (min … max):    1.283 s …  1.335 s    5 runs

**Comment by bddicken** [comment]
Thanks for contributing! Got Lua added in via https://github.com/bddicken/languages/pull/46 though, so I'm going to close this out.

**Pull request closed without merging**